### DISCUSSION TITLE
Fail to generate keys

### DISCUSSION THREAD

**Discussion by vinsonmixcare** [spatie/crypto#19] [Bugs]
Hi Thanks for this great package!

Got this error when generate keys

`
[$privateKey, $publicKey] = (new KeyPair())->generate();
`

`
Spatie\Crypto\Rsa\KeyPair::__construct(): Argument #1 ($digestAlgorithm) must be of type string, int given
`

PHP Version: 
`
PHP 8.1.5 (cli) (built: Apr 16 2022 00:03:52) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.1.5, Copyright (c) Zend Technologies
    with Xdebug v3.1.2, Copyright (c) 2002-2021, by Derick Rethans
    with Zend OPcache v8.1.5, Copyright (c), by Zend Technologies
`

**Comment by Jannik-Baar** [comment]
I encountered the same bug. It only happens in strict mode.

**Reply by freekmurze** [reply]
Feel free to PR a fix

**Reply by Jannik-Baar** [reply]
I just checked, the bug is already fixed in the main branch, happened in PR#20, but it never got released
